service/history/workflow/update/update.go

```diff
@@ -92,6 +92,7 @@ type (
 		acceptedEventID int64
 		onComplete      func()
 		instrumentation *instrumentation
+		admittedTime    time.Time
```

Comment:
I think this is perfectly fine; at the same time I was wondering if it would be better if we used a sequence number instead. The number could be assigned by the Registry (seems safe, right?). That way we could avoid - admittedly very rare and unlikely - issues where the machine time is drifting. (tbh, I've never had to deal with such an issue as I've never required such precision before from my application, so this is entirely hypothetical)

Reply:
I also feel that a sequence ID is more elegant than a timestamp but it's also a bit more code and a couple more ways it can go wrong. I don't think drift is a theoretical correctness concern here since we're always comparing stamps obtained from the same process and so only require that they are increasing?